ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/LabelClickSpikes.java

```diff
@@ -113,11 +112,12 @@ public void process(@Element KV<String, PubsubMessage> element, @Timestamp Insta
       if (timestamps.size() <= maxClicks) {
         out.output(element);
       } else {
-        PubsubMessage message = element.getValue();
-        Map<String, String> attributes = new HashMap<>(message.getAttributeMap());
-        addClickStatusToReportingUrlAttribute(attributes);
+        SponsoredInteraction interaction = element.getValue();
+        String reportingURL = addClickStatusToReportingUrlAttribute(interaction.reporterURL());
         ghostClickCounter.inc();
-        out.output(KV.of(element.getKey(), new PubsubMessage(message.getPayload(), attributes)));
+        out.output(KV.of(
+                element.getKey(),
+                SponsoredInteraction.builder().from(interaction).reporterURL(reportingURL).build()));
```

Comment:
Per other comment, this can become `interaction.toBuilder.reporterURL(reportingURL).build()`